src/utils/task-store.ts

```diff
@@ -0,0 +1,42 @@
+import { create } from 'zustand';
+import { persist } from 'zustand/middleware';
+import { LinkSecretStatus } from '~/components/Secrets/SecretsListView/SecretsListRowWithComponents';
+
+export type TaskStatus = `${LinkSecretStatus}`;
+
+export interface Task {
+  id: string;
+  status: TaskStatus;
+  error?: string;
+}
+
+interface TaskStore {
+  tasks: Record<string, Task>;
+  setTaskStatus: (id: string, status: TaskStatus, error?: string) => void;
+  clearTask: (id: string) => void;
+}
+
+export const useTaskStore = create<TaskStore>()(
+  persist(
+    (set) => ({
+      tasks: {},
+      setTaskStatus: (id, status, error) =>
+        set((state) => ({
+          tasks: {
+            ...state.tasks,
+            [id]: { id, status, error },
+          },
+        })),
+      clearTask: (id) =>
+        set((state) => {
+          // eslint-disable-next-line @typescript-eslint/no-unused-vars
+          const { [id]: _, ...rest } = state.tasks;
+          return { tasks: rest };
+        }),
+    }),
+    {
+      name: 'task-store', // save store to local storage
+      partialize: (state) => ({ tasks: state.tasks }),
+    },
```

Comment:
Why do we need to persist this in local storage?

Reply:
By default, React state and Zustand stores are in-memory only. Refreshing the page clears all states. Persisting this to localStorage allows the app to restore the state even after the user refreshes or closes and reopens the browser. 

i just expect users could feel more reliable and smooth with it and not wonder why the Error disappears just by refreshing pages.

Thinking about we would support 'Edit Secret' soon, users could edit the secrets with relinking to fix the errors.

Reply:
Great questions. i removed the local-storage and enjoy the annotation for the Error msg to ensure all users get the same status.